### ISSUE TITLE
jasmine_node_test hides test failure when missing imports

### ISSUE THREAD

**Issue by mattsoulanille** [bazelbuild/rules_nodejs#2438]
# 🐞 bug report
### Affected Rule

 The issue is caused by the rule: jasmine_node_test

### Is this a regression?

 Not sure. I haven't tested any other versions yet since I'm working in a new repo.

### Description
Under `bazel test`, `jasmine_node_test` treats the case where a test fails to run due to a missing module as a passing test run.

## 🔬 Minimal Reproduction
In [this repo](https://github.com/mattsoulanille/bazel-multiple-npm-packages-test/tree/d577bceacada100f0010e1ed7903aa5eba1624a0), run `bazel test foo_package:foo_package_test`, and the tests will pass, even though running `bazel run foo_package_foo_package_test` shows they fail to find the module `multiple_npm_packages/foo_package/subdir/bla`.

## 🔥 Exception or Error
[bazel run --define=VERBOSE_LOGS=1 foo_package:foo_package_test](https://gist.github.com/mattsoulanille/da14fc19c6d9d5f88c85de3b98f809ea)
[bazel test --define=VERBOSE_LOGS=1 foo_package:foo_package_test](https://gist.github.com/mattsoulanille/c1b14cd98a5624adc172513daa331d0b)

## 🌍  Your Environment

**Operating System:**

<pre>
  <code>
$ uname -a
Linux (computer name) 5.7.17-1rodete4-amd64 #1 SMP Debian 5.7.17-1rodete4 (2020-10-01) x86_64 GNU/Linux
  </code>
</pre>

**Output of `bazel version`:**

<pre>
  <code>
$ bazel version
Bazelisk version: v1.4.0
Build label: 4.0.0
Build target: bazel-out/k8-opt/bin/src/main/java/com/google/devtools/build/lib/bazel/BazelServer_deploy.jar
Build time: Thu Jan 21 07:33:24 2021 (1611214404)
Build timestamp: 1611214404
Build timestamp as int: 1611214404
  </code>
</pre>

**Rules_nodejs version: 3.0.0**

(Please check that you have matching versions between WORKSPACE file and `@bazel/*` npm packages.)

<pre>
  <code>
{
  "name": "bazel-multiple-npm-test",
  "version": "1.0.0",
  "main": "index.js",
  "license": "Apache-2.0",
  "devDependencies": {
    "@bazel/bazelisk": "^1.7.3",
    "@bazel/typescript": "^3.0.0",
    "@bazel/jasmine": "^3.0.0",
    "@types/jasmine": "^3.6.3",
    "jasmine": "^3.6.4",
    "typescript": "^4.1.3"
  }
}
  </code>
</pre>

**Anything else relevant?**
Aside from this bug, if you know how to fix the import error without resorting to `link_workspace_root`, i'm all ears. I didn't really understand [this section](https://github.com/bazelbuild/rules_nodejs/wiki#no-longer-link-the-workspace-root) in the 2.0 migration guide, since it sounds like I'd have to declare a pkg_npm rule for each ts_library rule I want to import, which doesn't seem right. [ts-api-guardian](https://github.com/angular/angular/blob/master/tools/ts-api-guardian/BUILD.bazel) seems to solve it by having a single ts_library that includes all of the project's sources, but I'm hoping to keep `bla.ts` in its own ts_library if possible. Edit: This question is likely related to https://github.com/bazelbuild/rules_nodejs/issues/2426

**Comment by mattsoulanille** [comment]
I think I've found an incremental build bug as well that persists even through `bazel clean`. 

Steps to reproduce:
1. Clone [the repo](https://github.com/mattsoulanille/bazel-multiple-npm-packages-test/tree/d577bceacada100f0010e1ed7903aa5eba1624a0) in the above bug.
2. Run `bazel run foo_package:foo_package_test`. Tests should fail with `Cannot find module 'multiple....'` as in the above bug report.
3. Edit `foo_package/BUILD.bazel` and add `link_workspace_root = True` to the `foo_package_test` rule. It should read:
```starlark
jasmine_node_test(
    name = "foo_package_test",
    srcs = [
        ":foo_test_lib",
    ],
    deps = [
        "//foo_package/subdir:bla_lib",
    ],
    link_workspace_root = True,
)
```
4. Run `bazel run foo_package:foo_package_test` again. Test should run and fail with `Expected 'foo bla bla bla' to equal 'wrong string'.`, and there should be no module import errors.
5. Revert `foo_package/BUILD.bazel` to its unedited state by removing the `link_workspace_root = True` arg from `jasmine_node_test`.
6. Run `bazel run foo_package:foo_package_test` a third time. Expected behavior would be for the tests to fail with the same `Cannot find module 'multiple....'` as in step 2, but instead, they fail with `Expected 'foo bla bla bla' to equal 'wrong string'.` as in step 4. (This still occurs if the test is edited to pass, I'm just using a test that's supposed to fail since it's one I've already shared in the above bug report).
7. Run `bazel clean` (optionally add `--expunge`, it doesn't change the result) and retry 6. The test still fails at `Expected 'foo bla bla bla' to equal 'wrong string'.` when it's expected to fail at `Cannot find module 'multiple....'`.
8. Remove `node_modules` and try step 6 again. Now, the test fails with `Cannot find module 'multiple....'` as expected.

I'm not sure what's causing this, but it seems to be related to how Bazel adds packages to node_modules.
